src/main/java/chopchop/model/Entry.java

```diff
@@ -0,0 +1,44 @@
+package chopchop.model;
+
+import static java.util.Objects.requireNonNull;
+
+import chopchop.model.attributes.Name;
+
+public abstract class Entry {
+    protected final Name name;
+
+    protected Entry(String name) {
+        requireNonNull(name);
+        this.name = new Name(name);
+    }
+
+    public String getName() {
+        return this.name.toString();
+    }
+
+    /**
+     * Returns true if both entries have the same name.
+     * This defines a weaker notion of equality between two entries.
+     */
+    public boolean isSame(Entry other) {
+        return other == this
+                || (other != null
+                && this.name.equals(other.name));
+    }
+
+    /**
+     * Returns true if both entries have the same fields.
+     * This defines a stronger notion of equality between two entries.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
isn't this just `this.isSame((Entry) other)`

Reply:
Removed.